addons/account/models/account_move.py

```diff
@@ -2042,11 +2042,11 @@ def _compute_show_reset_to_draft_button(self):
 
     @api.depends('company_id.account_fiscal_country_id', 'fiscal_position_id.country_id', 'fiscal_position_id.foreign_vat')
     def _compute_tax_country_id(self):
-        for record in self:
-            if record.fiscal_position_id.foreign_vat:
-                record.tax_country_id = record.fiscal_position_id.country_id
-            else:
-                record.tax_country_id = record.company_id.account_fiscal_country_id
+        foreign_vat_records = self.filtered(lambda r: r.fiscal_position_id.foreign_vat)
+        for record in foreign_vat_records:
```

Comment:
Is there a reason why you're not grouping here (with fiscal_position_id) if it improves that much if you group when you do with company_id?

Reply:
Yeah, most people would probably not have thousands of fiscal position, or often enough use the same